### ISSUE TITLE
Promises are not resolved 

### ISSUE THREAD

**Issue by luispastendev** [lukehaas/RunJS#380]
The code does not give the same results depending on the order of its execution 

error
```js
const obj = [
  {id: 1, title: 'foo',},
  {id: 2, title: 'bar',},
];
obj.filter((i) => i.id == 1);

const a = async function ajax() {
    const res = await fetch(`https://jsonplaceholder.typicode.com/todos/2`);
    const data = await res.json();
    console.log(data);
}
a();

fetch('https://jsonplaceholder.typicode.com/todos/3')
    .then(response => response.json())
    .then(console.log)
    .catch( console.warn );
```
![imagen](https://user-images.githubusercontent.com/4117596/149068090-0fc8b13d-c007-449e-83b2-fd9b43c22cc8.png)

success
```js
const a = async function ajax() {
    const res = await fetch(`https://jsonplaceholder.typicode.com/todos/2`);
    const data = await res.json();
    console.log(data);
}
a();

fetch('https://jsonplaceholder.typicode.com/todos/3')
    .then(response => response.json())
    .then(console.log)
    .catch( console.warn );

const obj = [
  {id: 1, title: 'foo',},
  {id: 2, title: 'bar',},
];
obj.filter((i) => i.id == 1);
```
![imagen](https://user-images.githubusercontent.com/4117596/149068179-5c927746-8893-436a-a769-dead49c69cd3.png)

**Comment by lukehaas** [comment]
Thanks @luispastendev. Try turning off the Runtime values setting, I think this is related to that:

![Screenshot 2022-01-12 at 09 38 33](https://user-images.githubusercontent.com/741344/149110458-f3b82101-8bd0-48ed-9ca9-98579562b017.png)

**Comment by lukehaas** [comment]
Thanks @luispastendev. All promises are resolving for me in both examples. What version of RunJS are you using?

**Comment by luispastendev** [comment]
I use version 2.2.2.0 with license

**Comment by lukehaas** [comment]
@luispastendev is it possible the response from the server was just slow?

**Comment by luispastendev** [comment]
it's just a test using this service https://jsonplaceholder.typicode.com/ ,but executing the following code only solves the last promise and the first one remains pending ... if you try to comment the execution of the filter both promises are resolved 

code: 
```
const obj = [
  {id: 1, title: 'foo',},
  {id: 2, title: 'bar',},
];
obj.filter((i) => i.id == 1);

const a = async function ajax() {
    const res = await fetch(`https://jsonplaceholder.typicode.com/todos/2`);
    const data = await res.json();
    console.log(data);
}
a();

fetch('https://jsonplaceholder.typicode.com/todos/3')
    .then(response => response.json())
    .then(console.log)
    .catch( console.warn );
```

**Comment by lukehaas** [comment]
Unfortunately, I'm seeing something completely different
![Screenshot 2022-01-12 at 17 49 35](https://user-images.githubusercontent.com/741344/149194576-fda6ca1b-7f2e-460c-85fc-a3a3e080e07a.png)

**Comment by luispastendev** [comment]
thats weird! I show you the version I have and I work with win 11 
![imagen](https://user-images.githubusercontent.com/4117596/149198464-fadb6df3-23ee-4059-a120-72f3c28b3ca6.png)

**Comment by oculus42** [comment]
I tried this on v2.3.1 and am able to consistently get three unique outputs with three minor variants:

**Variant 1: Async Wrapper**
```javascript
(async() => { const obj = [
    {id: 1, title: 'foo',},
    {id: 2, title: 'bar',},
  ];
  obj.filter((i) => i.id == 1);

  const a = async function ajax() {
    console.log('a start');
    const r1 = fetch(`https://jsonplaceholder.typicode.com/todos/2`)
    console.log('a fetch start');
    const res = await r1;
    console.log('a fetch complete');
    const data = await res.json();
    console.log('a parse');
    console.log(data);
  }
  a()

  fetch('https://jsonplaceholder.typicode.com/todos/3')
      .then(_ => (console.log('b  complete'), _))
      .then(response => response.json())
      .then(_ => (console.log('b parse'), _))
      .then(console.log)
      .catch( console.warn );
             
})();
```

**Variant 1: Output**
```text
'a start'
'a fetch start'
Promise { undefined }
'b  complete'
'b parse'
{
  userId: 1,
  id: 3,
  title: 'fugiat veniam minus',
  completed: false
}
```

**Variant 2: Non-async Wrapper**
```javascript
(() => { const obj = [
    {id: 1, title: 'foo',},
    {id: 2, title: 'bar',},
  ];
  obj.filter((i) => i.id == 1);

  const a = async function ajax() {
    console.log('a start');
    const r1 = fetch(`https://jsonplaceholder.typicode.com/todos/2`)
    console.log('a fetch start');
    const res = await r1;
    console.log('a fetch complete');
    const data = await res.json();
    console.log('a parse');
    console.log(data);
  }
  a()

  fetch('https://jsonplaceholder.typicode.com/todos/3')
      .then(_ => (console.log('b  complete'), _))
      .then(response => response.json())
      .then(_ => (console.log('b parse'), _))
      .then(console.log)
      .catch( console.warn );
             
})();
```
**Variant 2: Output**
```text
'a start'
'a fetch start'
'b  complete'
'b parse'
{
  userId: 1,
  id: 3,
  title: 'fugiat veniam minus',
  completed: false
}
```

Hope this helps.

**Variant3: Original (plus logging)**
```javascript
const obj = [
  {id: 1, title: 'foo',},
  {id: 2, title: 'bar',},
];
obj.filter((i) => i.id == 1);

const a = async function ajax() {
  console.log('a start');
  const r1 = fetch(`https://jsonplaceholder.typicode.com/todos/2`)
  console.log('a fetch start');
  const res = await r1;
  console.log('a fetch complete');
  const data = await res.json();
  console.log('a parse');
  console.log(data);
}
a()

fetch('https://jsonplaceholder.typicode.com/todos/3')
    .then(_ => (console.log('b  complete'), _))
    .then(response => response.json())
    .then(_ => (console.log('b parse'), _))
    .then(console.log)
    .catch( console.warn );
```

**Output**
```text
[ { id: 1, title: 'foo' } ]
'a start'
'a fetch start'
Promise { <pending> }
Promise { <pending> }
'b  complete'
'b parse'
{
  userId: 1,
  id: 3,
  title: 'fugiat veniam minus',
  completed: false
}
'a fetch complete'
'a parse'
{
  userId: 1,
  id: 2,
  title: 'quis ut nam facilis et officia qui',
  completed: false
}
```

**Comment by lukehaas** [comment]
@oculus42 it looks like all of these promises are being resolved unless I'm missing something?

**Comment by oculus42** [comment]
@lukehaas Variants 1 & 2 never log `a fetch complete` or anything else in that logic block, yet no errors are thrown, either. Something prevents that fetch from completing in the first two scenarios, but not the third.

All three outputs should include the `id: 2` object from the async function `a`.

**Comment by lukehaas** [comment]
@oculus42 I see now. You're seeing completely different output to what I see when I run these snippets. 
I see `a fetch complete` for all and `id: 2` for all.
Is there any possibility something external to RunJS is preventing these requests from completing?

**Comment by oculus42** [comment]
Using Telerik Fiddler on my PC, I am not always seeing the `todos/2` request being made. The `todos/3` request is always made.

**Comment by lukehaas** [comment]
Thanks @oculus42. That's very useful. So that I can narrow down my investigation a bit further, can you please answer the following:
Which version of Windows are you using?
Do you have 'Runtime values' (advanced preferences) - enabled?
Do you have transpilation enabled?

**Comment by oculus42** [comment]
@lukehaas Windows 10 Pro 21H2 (19044.1706) with RunJS 2.3.1.0.

Runtime values - show runtime values - Experimental is enabled
Transpilation is enabled.

I also experience this issue on macOS Monterey 12.3.1 with RunJS 2.3.1

If it helps I can try other systems/platforms. The two above are licensed; I have not tried on the free edition.

**Comment by lukehaas** [comment]
Do the requests also fail if you disabled runtime values?

**Comment by lukehaas** [comment]
@oculus42 I've now managed to observe this issue and I will work to get it fixed. Thanks for all the info you provided.

**Comment by lukehaas** [comment]
@oculus42 this should now be resolved in the latest version of RunJS - 2.4.1
